test/testdrive/mz-catalog.td

```diff
@@ -0,0 +1,53 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+> SHOW DATABASES
+Database
+-------------
+materialize
+
+> SELECT * FROM mz_catalog.mz_database
+global_id   database
+-----------------------
+1           materialize
+
+> CREATE DATABASE dummy
+
+> SHOW DATABASES
+Database
+------------
+materialize
+dummy
+
+> SELECT * FROM mz_catalog.mz_database
+global_id   database
+-----------------------
+1           materialize
+2           dummy
+
+! DROP DATABASE wrong_dummy
+unknown database
+
+> SELECT * FROM mz_catalog.mz_database
+global_id   database
+-----------------------
+1           materialize
+2           dummy
+
+> DROP DATABASE dummy
+
+> SHOW DATABASES
+Database
+-------------
+materialize
+
+> SELECT * FROM mz_catalog.mz_database
+global_id   database
+-----------------------
+1           materialize
```

Comment:
Would you mind to integrate this test into catalog.td instead? Basically anywhere there is a `SHOW DATABASES` call, stick a `SELECT * FROM mz_catalog.mz_databases` test right after it and check that the results are the same? Should save you some work in creating and dropping databases.